src/fhir_search/uri_parser.clj

```diff
@@ -0,0 +1,70 @@
+(ns fhir-search.uri-parser 
+  (:require [clojure.string :as str])
+  (:require [clojure.walk :refer [postwalk]])
+  (:import [java.net URI]))
+
+(defn gen-values [query]
+  (->> 
+   (str/split query #"&")
+   (reduce (fn [result element]
+             (let [first-group (first (str/split element #"=")) second-group (second (str/split element #"="))]
+               (conj result {:name (first (str/split first-group #":"))
+                             :modifiers (when (not (nil? (second (str/split first-group #":")))) 
+                                          (keyword (str "fhir.search.modifier/" (second (str/split first-group #":")))))
+                             :value second-group}))) [])))
+
+(defn get-path [string]
+  (let [path (remove empty? (-> (.getPath (URI. string)) (str/split #"/"))) structure {:type (last path)}]
```

Comment:
it is not easy to read all the bindings of the let because they are inline, can you declare one binding per line?

Reply:
why do you need to declare the `structure` binding?